src/linear_algebra/alternating.lean

```diff
@@ -0,0 +1,239 @@
+/-
+Copyright (c) 2020 Zhangir Azerbayev. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Eric Wieser, Zhangir Azerbayev
+-/
+
+import linear_algebra.multilinear
+import group_theory.perm.sign
+
+/-!
+# Alternating Maps
+
+We construct the bundled function `alternating_map`, which extends `multilinear_map` with all the
+arguments of the same type.
+
+## Main definitions
+* `alternating_map R M N ι` is the space of `R`-linear alternating maps from `ι → M` to `N`.
+* `f.map_eq_args` expresses that `f` is zero when two inputs are equal.
+* `f.map_swap` expresses that `f` is negated when two inputs are swapped.
+* `f.map_perm` expresses how `f` varies by a sign change under a permutation of its inputs.
+* An `add_comm_monoid`, `add_comm_group`, and `semimodule` structure over `alternating_map`s that
+  matches the definitions over `multilinear_map`s.
+
+## Implementation notes
+`alternating_map` is defined in terms of `map_eq_args`, as this is easier to work with than
+using `map_swap` as a definition, and does not require `has_neg N`.
+-/
+
+-- semiring / add_comm_monoid
+variables {R : Type*} [semiring R]
+variables {M : Type*} [add_comm_monoid M] [semimodule R M]
+variables {N : Type*} [add_comm_monoid N] [semimodule R N]
+
+-- semiring / add_comm_group
+variables {L : Type*} [add_comm_group L] [semimodule R L]
+
+-- ring / add_comm_group
+variables {R' : Type*} [ring R']
+variables {M' : Type*} [add_comm_group M'] [semimodule R' M']
+variables {N' : Type*} [add_comm_group N'] [semimodule R' N']
+
+variables {ι : Type*} [decidable_eq ι]
+
+set_option old_structure_cmd true
+
+section
+variables (R M N ι)
+/--
+An alternating map is a multilinear map that vanishes when two of its arguments are equal.
+-/
+structure alternating_map extends multilinear_map R (λ i : ι, M) N :=
+(map_eq_args' : ∀ (v : ι → M) (i j : ι) (h : v i = v j) (hij : i ≠ j), to_fun v = 0)
```

Comment:
```suggestion
(map_eq_zero' : ∀ (v : ι → M) (i j : ι) (h : v i = v j) (hij : i ≠ j), to_fun v = 0)
```
How do you like that?

Reply:
That reminds me a bit too much of `monoid_with_zero_hom.map_eq_zero`, which we might want to reuse for anisotropic multilinear maps.

Reply:
I see. Do you find `map_eq_zero_of_eq` too long?